app/Http/Controllers/Api/ProjectController.php

```diff
@@ -44,88 +59,152 @@ public function add_project(Request $request){
+                $extension = $file->getClientOriginalExtension();
+                $s3FileName = Str::uuid().'.'.$extension;
+                $file_type = explode('/', mime_content_type($file->getRealPath()))[0];
+                Storage::disk('s3')->put('Testfolder/'.$s3FileName, file_get_contents($file->getRealPath()));
+                $path = Storage::disk('s3')->url('Testfolder/'.$s3FileName);
+
+                $projectfile_entry = Projectfile::create([
+                    'project_id'=> $result->id,
+                    'file_type' => $file_type,
+                    'filename' => $s3FileName,
+                    'file_original_name' => $fileName,
+                    'file_extension' => $extension,
+                    'url' => $path
+                ]);
+            }
+
             return response()->json(['message'=>'Project saved successfully'],200);
         } catch(Exception $e){
+            DB::rollback();
             return response()->json($e->getMessage(),500);
         }
     }
 
     public function update_project(Request $request){
         $validator = Validator::make($request->all(), [
-            'project_id' => 'required|string',
+            'project_id' => 'required',
+            'images' => 'required|array',
+            'images.*' => 'required|max:'.(config('const.dropzone_max_file_size')*1024).'|mimetypes:'.config('const.dropzone_accepted_image'),
         ]);
         if ($validator->fails()) {
             return response()->json(['errors' => $validator->errors()], 422);
         }
         try{
             $project =Project::where('id',$request->project_id)->first();
-            $project->project_address_id = $request->project_address_id;
+            if (!$project) {
+                return response()->json(['message' => 'Project not found'], 404);
+            }
             $project->forename = $request->forename;
             $project->surname = $request->surname;
             $project->project_name = $request->project_name;
             $project->description = $request->description;
             $project->contact_mobile_no = $request->contact_mobile_no;
+            $project->contact_home_phone = $request->contact_home_phone;
             $project->contact_email = $request->contact_email;
+            $project->postcode = $request->postcode;
+            $project->county = $request->county;
+            $project->town = $request->town;
+            $project->categories = $request->categories;
+            $project->subcategories = $request->subcategories;
+            $project->notes = $request->notes;
             if(!$project->save()){
                 return response()->json(['message'=>'Unexpected error, please try after sometimes'],400);
             }
+            //delete images
+            $previousFiles = Projectfile::where('project_id', $project->id)->get();
+
+            foreach ($previousFiles as $previousFile) {
+                $s3Path = 'Testfolder/' . $previousFile->filename;
```

Comment:
Fetch folder name from const file

Reply:
Also you need to send the s3 file name here.

Reply:
resolved